### PR TITLE
fix(cosmos): crypto amount validation fixes

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

- add validation on zero crypto/fiat amount
- fix removing a one-char input defaulting to zero

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

N/A

## Testing

- Continue button should be disable on cosmos stake/unstake with 0
- Pressing the delete key on a single digit in staking should bring back empty input

### CODE REVIEW

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 1]
```diff
@@ -19,7 +19,8 @@ type StakingInputProps = {
 const cryptoInputValidation = {
   required: true,
   validate: {
-    validateCryptoAmount: (_: string) => {
+    validateCryptoAmount: (cryptoAmount: string) => {
+      if (bnOrZero(cryptoAmount).isZero()) return false
       // TODO: Implement when we have cosmos/osmosis balance data
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:24]: is this comment still valid? we def do have cosmos balance data?

**Review by 0xdef1cafe — COMMENTED** [review]
question

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/plugins/cosmos/components/StakingInput/StakingInput.tsx]: Nice catch, comment's gone as this is pretty much the implementation we needed here

**Review by 0xdef1cafe — APPROVED** [review]

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 2]
```diff
@@ -19,17 +19,18 @@ type StakingInputProps = {
...
 }
 const fiatInputValidation = {
   required: true,
   validate: {
-    validateFiatAmount: (_: string) => {
-      // TODO: Implement when we have cosmos/osmosis balance data
+    validateFiatAmount: (fiatAmount: string) => {
+      if (bnOrZero(fiatAmount).isZero()) return false
+
       return true
```
Comment (cjthompson) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:34]:
```suggestion
    validateFiatAmount: (fiatAmount: string) => Boolean(bn(fiatAmount).gt(0))
```

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 3]
```diff
@@ -19,17 +19,18 @@ type StakingInputProps = {
 const cryptoInputValidation = {
   required: true,
   validate: {
-    validateCryptoAmount: (_: string) => {
-      // TODO: Implement when we have cosmos/osmosis balance data
+    validateCryptoAmount: (cryptoAmount: string) => {
+      if (bnOrZero(cryptoAmount).isZero()) return false
       return true
     },
```
Comment (cjthompson) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:25]:
```suggestion
    validateCryptoAmount: (cryptoAmount: string) => Boolean(bn(cryptoAmount).gt(0))
```

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 4]
```diff
@@ -19,17 +19,18 @@ type StakingInputProps = {
...
 const fiatInputValidation = {
   required: true,
   validate: {
-    validateFiatAmount: (_: string) => {
-      // TODO: Implement when we have cosmos/osmosis balance data
+    validateFiatAmount: (fiatAmount: string) => {
+      if (bnOrZero(fiatAmount).isZero()) return false
+
       return true
     },
```
Comment (cjthompson) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:35]:
```suggestion
    validateFiatAmount: (fiatAmount: string) => Boolean(bn(fiatAmount).gt(0)),
```

**Review by cjthompson — COMMENTED** [review]
I'd like to see the validation function pulled out into something like `lib/math.ts` and renamed to `isGreaterThanZero` with tests added.

**Comment by gomesalexandre** [comment]
> I'd like to see the validation function pulled out into something like `lib/math.ts` and renamed to `isGreaterThanZero` with tests added.

Since the data that comes in is already sanitized, and we just return a call to `bignumber.js` API, wouldn't tests effectively just tests the BN API?
I can see the value of abstracting it into its own util however, but as a reusable crypto/fiat amount validation function: if it's going to live in `lib/math` and be named something general purpose like `isGreaterThanZero`, I'd probably just stick with the current implementation, else we end up rewriting `bignumber.js`

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 5]
```diff
@@ -19,19 +19,13 @@ type StakingInputProps = {
 const cryptoInputValidation = {
   required: true,
   validate: {
-    validateCryptoAmount: (_: string) => {
-      // TODO: Implement when we have cosmos/osmosis balance data
-      return true
-    },
+    validateCryptoAmount: (cryptoAmount: string) => Boolean(bn(cryptoAmount).gt(0)),
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:22]:
```suggestion
    validateCryptoAmount: (cryptoAmount: string) => bnOrZero(cryptoAmount).gt(0),
```

?

File: src/plugins/cosmos/components/StakingInput/StakingInput.tsx [hunk 6]
```diff
@@ -19,19 +19,13 @@ type StakingInputProps = {
...
   },
 }
 const fiatInputValidation = {
   required: true,
   validate: {
-    validateFiatAmount: (_: string) => {
-      // TODO: Implement when we have cosmos/osmosis balance data
-      return true
-    },
+    validateFiatAmount: (fiatAmount: string) => Boolean(bn(fiatAmount).gt(0)),
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/components/StakingInput/StakingInput.tsx:28]:
```suggestion
    validateFiatAmount: (fiatAmount: string) => bnOrZero(fiatAmount).gt(0),
```

**Review by 0xdef1cafe — APPROVED** [review]

**Review by theoboldfrazier — APPROVED** [review]

**Pull request merged**